### PR TITLE
feat: update wd-mon to work for OptimismPortal2

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Updates wd-mon so that it works properly with OptimismPortal2. Doesn't require any significant functional changes outside of the logic used to pick the starting block number. Used this as a chance to clean up the service and generalize it so it works better with chains other than OP Mainnet.

I've run this locally and it works. New API is a superset of the old API, just added some extra metrics and new options.

### CODE REVIEW

File: packages/chain-mon/src/wd-mon/constants.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * withdrawals by default. L1 block numbers here are based on the block number in which the
+ * OptimismPortal proxy contract was deployed to L1.
+ */
+export const DEFAULT_STARTING_BLOCK_NUMBERS: {
+  [ChainID in L2ChainID]?: number
+} = {
+  [L2ChainID.OPTIMISM]: 17365802 as const,
+  [L2ChainID.OPTIMISM_GOERLI]: 0 as const,
+  [L2ChainID.OPTIMISM_SEPOLIA]: 4071248 as const,
+  [L2ChainID.BASE_MAINNET]: 0 as const,
```
Comment (smartcontracts) [inline packages/chain-mon/src/wd-mon/constants.ts]: TODO: Need to fill this out before we merge

File: packages/common-ts/src/base-service/validators.ts [hunk 2]
```diff
@@ -49,6 +49,14 @@ const logLevel = makeValidator<LogLevel>((input) => {
   }
 })
 
+const address = makeValidator<string>((input) => {
+  if (!ethers.utils.isHexString(input, 20)) {
+    throw new Error(`expected input to be an address: ${input}`)
+  } else {
+    return input as string
```
Comment (roninjin10) [inline packages/common-ts/src/base-service/validators.ts]:
```suggestion
    return input as `0x${string}`
```
Cast it as this so the type works in wagmi/viem too

File: packages/chain-mon/src/wd-mon/constants.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { L2ChainID } from '@eth-optimism/sdk'
+
```
Comment (roninjin10) [inline packages/chain-mon/src/wd-mon/constants.ts:2]:
```suggestion
// TODO: consider moving to `@eth-optimism/constants` package and generating it from the superchain registry
// @see https://github.com/ethereum-optimism/optimism/pull/9041
```

**Review by roninjin10 — COMMENTED** [review]
lgtm made one suggestion about an Address typescript type and then happy to approve

File: packages/chain-mon/src/wd-mon/constants.ts (hunk 3 above)
Comment (roninjin10) [reply packages/chain-mon/src/wd-mon/constants.ts:2]: Bad link fixed it

Comment (smartcontracts) [reply packages/chain-mon/src/wd-mon/constants.ts:2]: Updated

File: packages/common-ts/src/base-service/validators.ts (hunk 2 above)
Comment (smartcontracts) [reply packages/common-ts/src/base-service/validators.ts]: Updated

**Review by roninjin10 — APPROVED** [review]

**Pull request merged**